tokio/tests/sync_rwlock.rs

```diff
@@ -103,7 +103,7 @@ fn write_read_shared_pending() {
     assert_ready!(t2.poll());
 
     let mut t3 = spawn(rwlock.write());
-    let mut _g2 = assert_pending!(t3.poll());
```

Comment:
What was the lint fired here? This looks like a false positive since it ensured `_g2` lived through the scope before.

Reply:
_g2 is a unit value here

Reply:
And clippy caught that? Neat. :)